### ISSUE TITLE
`context.getScope()` returns an wrong scope when `blockBindings: true`.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#3254]
I'm really surprised.

----

```
env:
    es6: true
ecmaFeatures:
    modules: true
rules:
    no-invalid-this: 2
```

```
> eslint -v
v1.0.0
> echo "function foo() { this; }" | eslint --stdin
>
```

No errors.
I expected an error, "Unexpected `this`."

Of course, this simple case exists in [its unit test](https://github.com/eslint/eslint/blob/master/tests/lib/rules/no-invalid-this.js#L108).
This problem happens only when `--env es6` was specified.

----

```
ecmaFeatures:
    modules: true
rules:
    no-invalid-this: 2
```

```
> eslint -v
v1.0.0
> echo "function foo() { this; }" | eslint --stdin

<text>
  1:19  error  Unexpected `this`  no-invalid-this

✖ 1 problem (1 error, 0 warnings)

>
```

It's OK if there is not `es6: true`.

----

If `--env es6` is specified, [here](https://github.com/eslint/eslint/blob/master/lib/rules/no-invalid-this.js#L297), `context.getScope()` returns the global scope. But this is always on `FunctionDeclaration` or `FunctionExpression`, so `context.getScope()` should return a function scope.

And the global scope is not strict mode, so this rule ignores the `this` keyword....

**Comment by mysticatea** [comment]
I could reproduce with:

```
ecmaFeatures:
    blockBindings: true
    modules: true
rules:
    no-invalid-this: 2
```


.....[here](https://github.com/eslint/eslint/blob/master/lib/eslint.js#L1043)?

**Comment by nzakas** [comment]
So `context. getScope` is returning the global scope inside of FunctionDeclaration?

**Comment by mysticatea** [comment]
Yes. Maybe it returns a function scope as expected in `BlockStatement` of `FunctionDeclaration.body`. But it's returning the containing scope at `FunctionDeclaration`, only under `blockBindings: true`.

I will work on this.
